src/wp-includes/blocks/template-part.php

```diff
@@ -109,16 +109,20 @@ function render_block_core_template_part( $attributes ) {
 	// is set in `wp_debug_mode()`.
 	$is_debug = WP_DEBUG && WP_DEBUG_DISPLAY;
 
-	if ( is_null( $content ) && $is_debug ) {
-		if ( ! isset( $attributes['slug'] ) ) {
-			// If there is no slug this is a placeholder and we dont want to return any message.
-			return;
+	if ( is_null( $content ) ) {
```

Comment:
Forgive me if I'm telling you something you already know but just in case - this change should be implemented upstream in the Gutenberg repo and backported to Core 🙇 

I think we may already be looking to do this in https://github.com/WordPress/gutenberg/pull/54354 ?